### PR TITLE
add another version of `listen()` for passing externally created `AsyncSocket` to `AsyncHttpServer`

### PR DESCRIPTION
**Pull request by derekdai**
To support, for example, socket activation services in systemd or inetd.

### CODE REVIEW

File: lib/pure/asynchttpserver.nim [hunk 1]
```diff
@@ -372,19 +372,25 @@ const
     ## This can be set on the command line during compilation
     ## via `-d:nimMaxDescriptorsFallback=N`
 
+proc listen*(server: AsyncHttpServer; socket: AsyncSocket) {.inline.} =
+  # Listen on a provided socket. This is caller's responsibility to
```
Comment (dom96) [inline lib/pure/asynchttpserver.nim]:
```suggestion
  # Listen on a provided socket. It is the caller's responsibility to
```

File: lib/pure/asynchttpserver.nim [hunk 2]
```diff
@@ -372,19 +372,25 @@ const
...
+  server.socket.listen()
+
 proc listen*(server: AsyncHttpServer; port: Port; address = ""; domain = AF_INET) =
   ## Listen to the given port and address.
   when declared(maxDescriptors):
     server.maxFDs = try: maxDescriptors() except: nimMaxDescriptorsFallback
   else:
     server.maxFDs = nimMaxDescriptorsFallback
-  server.socket = newAsyncSocket(domain)
+  let socket = newAsyncSocket(domain)
```
Comment (dom96) [inline lib/pure/asynchttpserver.nim]: I don't think this will work. The `server.socket` will never be set if this proc is called.

**Review by dom96 — CHANGES_REQUESTED** [review]

File: lib/pure/asynchttpserver.nim (hunk 2 above)
Comment (derekdai) [reply lib/pure/asynchttpserver.nim]: You are right.

File: lib/pure/asynchttpserver.nim (hunk 1 above)
Comment (derekdai) [reply lib/pure/asynchttpserver.nim]: Done.

**Pull request closed without merging**